### PR TITLE
Epmlstrtai 748

### PR DESCRIPTION
**Pull request by Andrewisnew**
- Added relation between step and step suggestion 
- Edited create, update, delete functionality

### CODE REVIEW

File: bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java [hunk 1]
```diff
@@ -108,20 +120,26 @@ public StepDTO addStepToCase(Long projectId, Long suitId, Long caseId,
...
-        stepDAO.save(step);
-
+        Step oldStep = stepDAO.findById(stepId).orElseThrow(NotFoundException::new);
+        throwExceptionIfStepIsNotInCase(caze, oldStep);
+        StepSuggestion oldStepSuggestion = stepSuggestionService.getStepSuggestion(projectId, oldStep);
+        Step newStep = stepTransformer.updateFromDto(stepUpdateDTO, oldStep);
+        StepSuggestion newStepSuggestion = stepSuggestionService.getStepSuggestion(projectId, newStep);
+        stepDAO.save(newStep);
+        if(!Objects.equals(oldStepSuggestion, newStepSuggestion)) {
+            oldStepSuggestion.getSteps().remove(oldStep);
```
Comment (AleksandrKugushev) [inline bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java]: In this case, you abuse getSteps() excess functionality, so if you want to make a refactoring in the future, you'll have to reimplement full functionality of getSteps() return value. I suggest using 'remove' and 'add' methods of StepSuggestion: 
```
newStepSuggestion.remove(oldStep);
newStepSuggestion.remove(newStep);
```

File: bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java [hunk 2]
```diff
@@ -138,15 +156,39 @@ public void removeStep(Long projectId, Long suitId, Long caseId, Long stepId) {
...
     }
 
+    public void removeStep(Long projectId, Long stepId) {
+        Project project = checkNotNull(projectService.getProjectByProjectId(projectId));
+
+        for (Suit suit: project.getSuits()) {
+            for (Case caze: suit.getCases()) {
+                for (Step step: caze.getSteps()) {
+                    if(stepId.equals(step.getId())){
+                        caze.removeStep(step);
```
Comment (AleksandrKugushev) [inline bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java]: What about ConcurrentModificationException? Can you check it? https://stackoverflow.com/questions/10431981/remove-elements-from-collection-while-iterating/10432084

Comment (Andrewisnew) [reply bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java]: There is return statement after removing a step, so iterator won't take new element after collection modification and ConcurrentModificationException won't be thrown

File: bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java (hunk 1 above)
Comment (Andrewisnew) [reply bdd-generator/src/main/java/com/epam/test_generator/services/StepService.java]: Fixed

**Pull request merged**